### PR TITLE
add query builder doc strings

### PR DESCRIPTION
**Pull request by WillMcDGSK**
@paltusplintus @JonathanDeacon Please can one of you review!
@PranjaliParnerkar FYI and also

Adding doc strings to the majority of functions in the QueryBuilder class

### CODE REVIEW

File: query_builders/query_builder.py [hunk 1]
```diff
@@ -328,7 +361,18 @@ def check_connectedness(labels: list, rels: list):
             return False
 
     @staticmethod
-    def enrich_labels_from_rels(labels: list, rels: list, oclass_marker: str):
+    def enrich_labels_from_rels(labels: list, rels: list, oclass_marker: str) -> list:
+        """
+        Add any labels found in relationships that are not already present in labels.
+        As sometimes a label is not optional when a relationship it is involved in IS optional, 
+        we decide on whether to make that label optional.
```
Comment (JonathanDeacon) [inline query_builders/query_builder.py:368]: I found this wording a little bit confusing. Perhaps it should be something along the lines of:

Extends the labels list with any labels that are only present in relationships. Conditionally non optional labels may become optional during this process if they are involved in an relationship that is optional.

**Review by JonathanDeacon — CHANGES_REQUESTED** [review]
Looks good! A few minor comments from me:

File: query_builders/query_builder.py (hunk 1 above)
Comment (WillMcDGSK) [reply query_builders/query_builder.py:368]: Sounds much better, thanks!

**Comment by WillMcDGSK** [comment]
@JonathanDeacon Thanks for the comments, I've added the changes you requested!

**Review by JonathanDeacon — APPROVED** [review]
Looks good to me!

**Pull request merged**